tests/ga/test_update.py

```diff
@@ -2869,7 +2814,65 @@ def test_it_should_process_goal_state_only_on_new_goal_state(self):
             self.assertEqual(2, remote_access_handler.run.call_count, "remote_access_handler.run() should have been called on a new goal state")
 
 
-class ReportStatusTestCase(AgentTestCase):
+class HeartbeatTestCase(AgentTestCase):
+
+    @patch("azurelinuxagent.common.logger.info")
+    @patch("azurelinuxagent.ga.update.add_event")
+    def test_telemetry_heartbeat_creates_event(self, patch_add_event, patch_info, *_):
+        
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as mock_protocol:
```

Comment:
This test originally used `mock_protocol = WireProtocol('foo.bar')`, but I wanted to test the hook for IMDS network calls anyway, so I switched it to using the newer way of doing things.